<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Evangelyn Sim" />


<title>Processing fastq files to generate sequencing read count matrix</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="site_libs/anchor-sections-1.0/anchor-sections.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Human_Development_ATACseq_bulk</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/evangelynsim/Human_Development_ATACseq_bulk">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Processing fastq files to generate sequencing read count matrix</h1>
<h4 class="author">Evangelyn Sim</h4>
<h4 class="date">2/15/2021</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-02-19
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Human_Development_ATACseq_bulk/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210216code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210216)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210216code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210216)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomevangelynsimHumanDevelopmentATACseqbulktree294d8302e6f1302302285058851a249e73804868targetblank294d830a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/evangelynsim/Human_Development_ATACseq_bulk/tree/294d8302e6f1302302285058851a249e73804868" target="_blank">294d830</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomevangelynsimHumanDevelopmentATACseqbulktree294d8302e6f1302302285058851a249e73804868targetblank294d830a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/evangelynsim/Human_Development_ATACseq_bulk/tree/294d8302e6f1302302285058851a249e73804868" target="_blank">294d830</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  *.noXYMT.bed.tidy.bed
    Untracked:  *xls.bg.bed
    Untracked:  *xls.dn.bed
    Untracked:  *xls.up.bed
    Untracked:  Development_noXY.jn.rnk
    Untracked:  FetalvsYoung_noXY.jn.rnk
    Untracked:  Homo_sapiens.GRCh38.96.fulllength.saf
    Untracked:  YoungvsAdult_noXY.jn.rnk
    Untracked:  analysis/*.dn.bed.homeranno.txt
    Untracked:  analysis/*.up.bed.homeranno.txt
    Untracked:  analysis/00.WorkFlowR_setting.R
    Untracked:  code/EnDrich.R
    Untracked:  code/EnDrichProc_Development_noXY.R
    Untracked:  code/EnDrichProc_FetalvsYoung_noXY.R
    Untracked:  code/EnDrichProc_YoungvsAdult_noXY.R
    Untracked:  header.sam
    Untracked:  humanATAC*bed.saf
    Untracked:  humanATAC*bed.saf.pe.q30.mx
    Untracked:  humanATAC*bed.saf.pe.q30.mx.all
    Untracked:  humanATAC*bed.saf.pe.q30.mx.all.fix
    Untracked:  humanATAC*bed.saf.pe.q30.mx.chr
    Untracked:  humanATAC*bed.saf.pe.q30.mx.fix
    Untracked:  humanATAC*bed.saf.pe.q30.mx.hum.fix
    Untracked:  output/20190801_ATAC_samplesheet.txt
    Untracked:  output/ATACseq_samplesheet.txt
    Untracked:  output/atac_hum_tss_pe_mapk30_q30.mx.all_unfiltered.csv
    Untracked:  output/atac_hum_tss_pe_mapk30_q30.mx.chr
    Untracked:  output/atac_hum_tss_pe_mapk30_q30.mx.hum.fix_filt.csv
    Untracked:  output/humanATAC_peaks_cov2_rmBL.bed.saf.pe.q30.mx.MvsF.fix_filt.csv
    Untracked:  output/humanATAC_peaks_cov2_rmBL.bed.saf.pe.q30.mx.all.fix_filt.csv
    Untracked:  output/humanATAC_peaks_cov2_rmBL.bed.saf.pe.q30.mx.all_unfiltered.csv
    Untracked:  output/humanATAC_peaks_cov2_rmBL.bed.saf.pe.q30.mx.hum.fix_filt.csv
    Untracked:  output/logCPM_humanATAC_peaks_cov2_rmBL.bed.saf.pe.q30.mx.all.fix_filt.csv

Unstaged changes:
    Modified:   analysis/about.Rmd
    Modified:   analysis/index.Rmd
    Modified:   analysis/license.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/02.Primary_Analysis.Rmd</code>) and HTML (<code>docs/02.Primary_Analysis.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/evangelynsim/Human_Development_ATACseq_bulk/blob/294d8302e6f1302302285058851a249e73804868/analysis/02.Primary_Analysis.Rmd" target="_blank">294d830</a>
</td>
<td>
evangelynsim
</td>
<td>
2021-02-19
</td>
<td>
wflow_publish(c(“analysis/01.Generate_reference_genome.Rmd”,
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>Following sequencing and obtaining .fastq.gz file, the first step is to perform trimming and mapping of the sequencing data to generate bam files. All these steps were performed using bash code.</p>
<p>Bam files were then used for removal of duplicated and low quality (&lt;Q30) reads and subsequently subjected to read counting to generate a count matrix.</p>
<p>Human developmental bulk ATAC-seq were performed using paired-end sequencing method and below are the scripts for primary processing of paired-end sequencing read.</p>
<div id="used-libraries-and-functions" class="section level3">
<h3>Used libraries and functions</h3>
<ul>
<li>skewer/0.2.2</li>
<li>bwa/0.7.17</li>
<li>samtools/1.8</li>
<li>parallel</li>
<li>subread/1.5.0</li>
<li>bedtools/2.27.1</li>
<li>macs14</li>
<li>bedops/2.4.20</li>
</ul>
</div>
</div>
<div id="basic-processing-of-all-samples" class="section level1">
<h1>1. Basic processing of all samples</h1>
<div id="trimming-of-sequencing-read" class="section level3">
<h3>Trimming of sequencing read</h3>
<pre><code>#!/bin/bash

# function to run skewer quality trimming
runskew(){
FQZ1=$1
FQZ2=`echo $FQZ1 | sed &#39;s/_R1.fastq.gz/_R2.fastq.gz/&#39;`
skewer -t 8 -q 20 $FQZ1 $FQZ2
}
export -f runskew

# actually run skewer
parallel -j3 runskew ::: *_R1.fastq.gz</code></pre>
</div>
<div id="mapping-of-skewer-trimmed-.fastq-to-mouse-reference-genome-using-bwa" class="section level3">
<h3>Mapping of Skewer trimmed .fastq to mouse reference genome using BWA</h3>
<pre><code>runbwamempe() {
FQ1=$1
FQ2=`echo $FQ1 | sed &#39;s/R1.fastq-trimmed-pair1.fastq/R1.fastq-trimmed-pair2.fastq/&#39;`
BASE=`echo $FQ1 | sed &#39;s/_R1.fastq-trimmed-pair1.fastq//&#39;`

REF=/group/card2/Evangelyn_Sim/Transcriptome_chromatin_human/Sequencing_ATAC_RNA/refgenome/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa

bwa mem -t 20 $REF $FQ1 $FQ2 \
| samtools view -uSh - \
| samtools sort -@10 -o ${BASE}.sort.bam
samtools index ${BASE}.sort.bam

samtools flagstat ${BASE}.sort.bam &gt; ${BASE}.sort.bam.stats
}
export -f runbwamempe


# actually run bwa pe
ls *_R1.fastq-trimmed-pair1.fastq | parallel -u -j4 runbwamempe {}</code></pre>
</div>
<div id="remove-duplicated-reads" class="section level3">
<h3>Remove duplicated reads</h3>
<pre><code>#!/bin/bash

nodup(){
BAM=$1
OUT=`echo $BAM | sed &#39;s/.bam/_nodup.bam/&#39;`
samtools rmdup $BAM $OUT
}
export -f nodup
parallel nodup ::: `ls *bam | grep -v dup`</code></pre>
</div>
<div id="merge-replicated-.bam-files." class="section level3">
<h3>Merge replicated .bam files.</h3>
<p>Make a directory called “merged” and ln all .bam files to the folder and perform the following.</p>
<pre class="bash"><code>#!/bin/bash

samtools view -H `ls *bam | head -1` &gt; header.sam
for BASE in `ls *bam | cut -d &#39;_&#39; -f1 | sort -u ` ; do
  rm $BASE.mg.bam
  samtools merge -h header.sam $BASE.mg.bam *${BASE}*bam &amp;
done
wait
</code></pre>
<pre><code>ls: cannot access *bam: No such file or directory
bash: line 2: samtools: command not found
ls: cannot access *bam: No such file or directory</code></pre>
</div>
<div id="remove-low-quality-q30-sequencing-reads" class="section level3">
<h3>Remove low quality (Q&lt;30) sequencing reads</h3>
<pre><code>#!/bin/bash

# function to run bwa in paired end mode
runsamtoolsmapq() {
BAM=$1

NAME=`echo ${BAM} | sed &#39;s/.bam//&#39;`
REF=/group/card2/Evangelyn_Sim/Transcriptome_chromatin_human/Sequencing_ATAC_RNA/refgenome/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa

samtools view -q 30 -f 0x2 -b -h ${BAM} &gt; ${NAME}.mapq30.bam

wait

samtools index ${NAME}.mapq30.bam
samtools flagstat ${BAM}.mapq30.bam &gt; ${BAM}.mapq30.bam.flagstats
samtools idxstats ${BAM}.mapq30.bam &gt; ${BAM}.mapq30.bam.idxstats

}
export -f runsamtoolsmapq


# actually run runsamtoolsmapq pe 
ls *.bam | parallel -u -j5 runsamtoolsmapq {}</code></pre>
</div>
<div id="investigate-the-distribution-of-reads-on-different-chromosome" class="section level3">
<h3>Investigate the distribution of reads on different chromosome</h3>
<pre><code>#!/bin/bash

cntrds(){
BAM=$1
samtools view -q30 $BAM | cut -f3 | sort -T .| uniq -c | sed &quot;s/^/${BAM}/&quot;
}
export -f cntrds

ls *bam | parallel cntrds {} | tee -a read_counts2_q30.txt</code></pre>
</div>
</div>
<div id="kbp---tss" class="section level1">
<h1>2. 1kbp +/- TSS</h1>
<div id="counting-reads-from-bam-files-across-human-reference-genome-for-1kbp---tss-to-identify-associated-genes" class="section level3">
<h3>Counting reads from bam files across human reference genome for 1kbp +/- TSS to identify associated genes</h3>
<pre class="bash"><code>#!/bin/bash

SAF=/group/card2/Evangelyn_Sim/Transcriptome_chromatin_human/Sequencing_ATAC_RNA/refgenome/tss_1kbp.saf
OUTPE=atac_hum_tss_pe_mapk30_q30.mx

#featureCounts -p -Q 30 -T 20 -s 0 -a $SAF -F SAF -o $OUTPE *bam
</code></pre>
</div>
<div id="tidy-1kbp---tss-count-matrix" class="section level3">
<h3>Tidy 1kbp +/- TSS count matrix</h3>
<pre class="bash"><code>
#!/bin/bash

for MX in `ls *mx` ; do
   sed 1d $MX | sed &#39;s/.mg.mapq30.bam//g&#39; &gt; $MX.all
   sed 1d $MX | cut -f1-6 | sed &#39;s/.mg.mapq30.bam//g&#39; &gt; $MX.chr
   sed 1d $MX | cut -f1,7- | sed &#39;s/.mg.mapq30.bam//g&#39; &gt; $MX.all.fix
   sed 1d $MX | cut -f1,7-26 | sed &#39;s/.mg.mapq30.bam//g&#39; &gt; $MX.hum.fix
   
done
wait
</code></pre>
</div>
<div id="filter-out-low-counts-genes-from-1kbp---tss-count-matrix" class="section level3">
<h3>Filter out low counts genes from 1kbp +/- TSS count matrix</h3>
<p>Filtering out low counts genes by running the following filter.sh as</p>
<p>bash filter.sh hrna_dev_mf_fulllen_se_strrev_q30.mx.all.fix</p>
<p>filter.sh</p>
<pre><code>head -1 $1 &gt; ${1}_filt
awk &#39;{
  min = max = sum = $2;       # Initialize to the first value (2nd field)
  sum2 = $2 * $2              # Running sum of squares
  for (n=3; n &lt;= NF; n++) {   # Process each value on the line
    if ($n &lt; min) min = $n    # Current minimum
    if ($n &gt; max) max = $n    # Current maximum
    sum += $n;                # Running sum of values
    sum2 += $n * $n           # Running sum of squares
  }
  print sum/(NF-1) ;
}&#39; $1 &gt; avg
paste avg $1 | awk &#39;$1 &gt;= 10&#39; | cut -f2- | tr &#39; &#39; &#39;\t&#39; &gt;&gt; ${1}_filt
rm avg</code></pre>
</div>
</div>
<div id="peaks" class="section level1">
<h1>3. Peaks</h1>
<div id="peak-call-of-individual-sample" class="section level3">
<h3>Peak call of individual sample</h3>
<pre><code>#!/bin/bash
BAMS=&#39;*bam&#39;
BASENAME=humanATAC
PEAKBED=${BASENAME}_peaks.bed
PEAKSAF=${BASENAME}_peaks.saf
OUT=${BASENAME}_pks.txt
MX=${BASENAME}_pks_se.mx

PATH=$PATH:/usr/local/installed/macs/1.4.2-1/python-2.7.11/.//bin/

ls $BAMS | parallel macs14 -t {} -n {}_macs

done

exit</code></pre>
</div>
<div id="curate-peaks-that-exist-in-more-than-2-or-3-samples-to-form-a-peak-set" class="section level3">
<h3>Curate peaks that exist in more than 2 or 3 samples to form a peak set</h3>
<pre><code>for BED in *peaks.bed ; do
 awk &#39;{OFS=&quot;\t&quot;} {if ($2&lt;1) print $1,1,$3 ; else print $0 }&#39; $BED | awk &#39;NF==&quot;5&quot;&#39;&gt; tmp
 mv tmp $BED
done

rm humanATAC_peaks_cov*.bed
for COV in 2 3 ; do
  bedtools multiinter -i *_macs_peaks.bed \
 | cut -f-4 | awk -v C=$COV &#39;$4&gt;=C &amp;&amp; NF==4&#39; \
 | bedtools merge -i - &gt; humanATAC_peaks_cov${COV}.bed
done

exit</code></pre>
</div>
<div id="remove-blacklisted-peaks" class="section level3">
<h3>Remove blacklisted peaks</h3>
<pre><code>#!/bin/bash

wget https://github.com/Boyle-Lab/Blacklist/blob/master/lists/Blacklist_v1/hg38-blacklist.bed.gz

cat hg38.blacklist.bed | sed &#39;s/chr//&#39; &gt; hg38.blacklist.tidy.bed

BL=/group/card2/Evangelyn_Sim/Transcriptome_chromatin_human/Sequencing_ATAC_RNA/refgenome/hg38.blacklist.tidy.bed

bedops -n -1 humanATAC_peaks_cov2.bed $BL  &gt; humanATAC_peaks_cov2_rmBL.bed

bedops -n -1 humanATAC_peaks_cov3.bed $BL  &gt; humanATAC_peaks_cov3_rmBL.bed</code></pre>
</div>
<div id="count-individual-sample-to-the-curated-peak-set" class="section level3">
<h3>Count individual sample to the curated peak set</h3>
<pre class="bash"><code>#!/bin/bash

for BED in humanATAC*bed ; do

  SAF=$BED.saf
  OUT=$SAF.pe.q30.mx
  awk &#39;{OFS=&quot;\t&quot;} {print &quot;PK&quot;NR&quot;_&quot;$1&quot;:&quot;$2&quot;-&quot;$3,$1,$2,$3,&quot;+&quot;}&#39; $BED &gt; $SAF
  ( featureCounts -p -Q 30 -T 20 -s 0 -a $SAF -F SAF -o $OUT *bam
  sed 1d $OUT | cut -f1,7- &gt; tmp ; mv tmp $OUT ) &amp;

done
</code></pre>
<pre><code>awk: fatal: cannot open file `humanATAC*bed&#39; for reading (No such file or directory)
bash: line 7: featureCounts: command not found</code></pre>
</div>
<div id="tidy-peak-count-matrix" class="section level3">
<h3>Tidy peak count matrix</h3>
<pre class="bash"><code>#!/bin/bash

for MX in `ls *mx` ; do
   cat $MX | sed &#39;s/.mg.mapq30.bam//g&#39;  &gt; $MX.fix
   cat $MX | cut -f1-21 | sed &#39;s/.mg.mapq30.bam//g&#39; &gt; $MX.hum.fix
done
wait
</code></pre>
</div>
<div id="filter-out-low-counts-genes-from-peak-count-matrix" class="section level3">
<h3>Filter out low counts genes from peak count matrix</h3>
<p>Filtering out low counts genes by running the following filter.sh as</p>
<p>bash filter.sh hrna_dev_mf_fulllen_se_strrev_q30.mx.all.fix</p>
<p>filter.sh</p>
<pre><code>head -1 $1 &gt; ${1}_filt
awk &#39;{
  min = max = sum = $2;       # Initialize to the first value (2nd field)
  sum2 = $2 * $2              # Running sum of squares
  for (n=3; n &lt;= NF; n++) {   # Process each value on the line
    if ($n &lt; min) min = $n    # Current minimum
    if ($n &gt; max) max = $n    # Current maximum
    sum += $n;                # Running sum of values
    sum2 += $n * $n           # Running sum of squares
  }
  print sum/(NF-1) ;
}&#39; $1 &gt; avg
paste avg $1 | awk &#39;$1 &gt;= 10&#39; | cut -f2- | tr &#39; &#39; &#39;\t&#39; &gt;&gt; ${1}_filt
rm avg</code></pre>
</div>
</div>
<div id="group-analysis" class="section level1">
<h1>4. Group analysis</h1>
<div id="merge-.bam-files-from-the-same-group-for-transcription-factor-motif-analysis-and-igv-visualization-of-peaks" class="section level3">
<h3>Merge .bam files from the same group for transcription factor motif analysis and IGV visualization of peaks</h3>
<pre class="bash"><code>#!/bin/bash

# mkdir called merged then ln all *.bam to the folder

# then run samtools merge on the bam files
samtools view -H `ls *bam | head -1` &gt; header.sam
for BASE in `ls *bam | cut -d &#39;_&#39; -f1 | sort -u ` ; do
  rm $BASE.mg.bam
  samtools merge -h header.sam $BASE.mg.bam ${BASE}*bam &amp;
done
wait
</code></pre>
<pre><code>ls: cannot access *bam: No such file or directory
bash: line 5: samtools: command not found
ls: cannot access *bam: No such file or directory</code></pre>
</div>
<div id="index-merged-.bam-files" class="section level3">
<h3>Index merged .bam files</h3>
<pre><code>#!/bin/bash

# function to run bwa in paired end mode
runbamindex() {
BAM=$1

samtools index $BAM
}
export -f runbamindex

ls *.bam | parallel -u -j4 runbamindex {}</code></pre>
</div>
<div id="macs-peak-calling-and-then-remove-blacklisted-peaks-method-same-as-above" class="section level3">
<h3>MACS peak calling and then remove blacklisted peaks (method same as above)</h3>
</div>
<div id="remove-blacklisted-peaks-1" class="section level3">
<h3>### Remove blacklisted peaks</h3>
<pre><code>#!/bin/bash

BL=/group/card2/Evangelyn_Sim/Transcriptome_chromatin_human/Sequencing_ATAC_RNA/refgenome/hg38.blacklist.tidy.bed

for BED in *_macs_peaks.bed ; do
   bedops -n -1 $BED $BL  &gt; $BED.rmBL.bed
done</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS:   /hpc/software/installed/R/3.6.1/lib64/R/lib/libRblas.so
LAPACK: /hpc/software/installed/R/3.6.1/lib64/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] workflowr_1.6.2

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.5       rstudioapi_0.11  whisker_0.4      knitr_1.30      
 [5] magrittr_1.5     R6_2.5.0         rlang_0.4.7      stringr_1.4.0   
 [9] tools_3.6.1      xfun_0.18        git2r_0.27.1     htmltools_0.5.0 
[13] ellipsis_0.3.1   rprojroot_1.3-2  yaml_2.2.1       digest_0.6.27   
[17] tibble_3.0.3     lifecycle_0.2.0  crayon_1.3.4     later_1.1.0.1   
[21] vctrs_0.3.2      promises_1.1.1   fs_1.5.0         glue_1.4.2      
[25] evaluate_0.14    rmarkdown_2.5    stringi_1.5.3    compiler_3.6.1  
[29] pillar_1.4.6     backports_1.1.10 httpuv_1.5.4     pkgconfig_2.0.3 </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
