---
title: "Processing fastq files to generate sequencing read count matrix"
author: "Evangelyn Sim"
date: "2/15/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```




## Introduction

Following sequencing and obtaining .fastq.gz file, the first step is to perform trimming and mapping of the sequencing data to generate bam files. All these steps were performed using bash code. 

Bam files were then used for removal of duplicated and low quality (<Q30) reads and subsequently subjected to read counting to generate a count matrix.

Human developmental bulk ATAC-seq were performed using paired-end sequencing method and below are the scripts for primary processing of paired-end sequencing read.


### Used libraries and functions

* skewer/0.2.2
* bwa/0.7.17
* samtools/1.8
* parallel
* subread/1.5.0
* bedtools/2.27.1
* macs14
* bedops/2.4.20


# 1. Basic processing of all samples

### Trimming of sequencing read

    #!/bin/bash

    # function to run skewer quality trimming
    runskew(){
    FQZ1=$1
    FQZ2=`echo $FQZ1 | sed 's/_R1.fastq.gz/_R2.fastq.gz/'`
    skewer -t 8 -q 20 $FQZ1 $FQZ2
    }
    export -f runskew

    # actually run skewer
    parallel -j3 runskew ::: *_R1.fastq.gz


### Mapping of Skewer trimmed .fastq to mouse reference genome using BWA

    runbwamempe() {
    FQ1=$1
    FQ2=`echo $FQ1 | sed 's/R1.fastq-trimmed-pair1.fastq/R1.fastq-trimmed-pair2.fastq/'`
    BASE=`echo $FQ1 | sed 's/_R1.fastq-trimmed-pair1.fastq//'`

    REF=/group/card2/Evangelyn_Sim/Transcriptome_chromatin_human/Sequencing_ATAC_RNA/refgenome/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa

    bwa mem -t 20 $REF $FQ1 $FQ2 \
    | samtools view -uSh - \
    | samtools sort -@10 -o ${BASE}.sort.bam
    samtools index ${BASE}.sort.bam

    samtools flagstat ${BASE}.sort.bam > ${BASE}.sort.bam.stats
    }
    export -f runbwamempe


    # actually run bwa pe
    ls *_R1.fastq-trimmed-pair1.fastq | parallel -u -j4 runbwamempe {}


### Remove duplicated reads

    #!/bin/bash

    nodup(){
    BAM=$1
    OUT=`echo $BAM | sed 's/.bam/_nodup.bam/'`
    samtools rmdup $BAM $OUT
    }
    export -f nodup
    parallel nodup ::: `ls *bam | grep -v dup`


### Merge replicated .bam files.

Make a directory called "merged" and ln all .bam files to the folder and perform the following.

```{bash}
#!/bin/bash

samtools view -H `ls *bam | head -1` > header.sam
for BASE in `ls *bam | cut -d '_' -f1 | sort -u ` ; do
  rm $BASE.mg.bam
  samtools merge -h header.sam $BASE.mg.bam *${BASE}*bam &
done
wait


```

### Remove low quality (Q<30) sequencing reads

    #!/bin/bash

    # function to run bwa in paired end mode
    runsamtoolsmapq() {
    BAM=$1

    NAME=`echo ${BAM} | sed 's/.bam//'`
    REF=/group/card2/Evangelyn_Sim/Transcriptome_chromatin_human/Sequencing_ATAC_RNA/refgenome/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa

    samtools view -q 30 -f 0x2 -b -h ${BAM} > ${NAME}.mapq30.bam

    wait

    samtools index ${NAME}.mapq30.bam
    samtools flagstat ${BAM}.mapq30.bam > ${BAM}.mapq30.bam.flagstats
    samtools idxstats ${BAM}.mapq30.bam > ${BAM}.mapq30.bam.idxstats

    }
    export -f runsamtoolsmapq


    # actually run runsamtoolsmapq pe 
    ls *.bam | parallel -u -j5 runsamtoolsmapq {}
    
    

### Investigate the distribution of reads on different chromosome

    #!/bin/bash

    cntrds(){
    BAM=$1
    samtools view -q30 $BAM | cut -f3 | sort -T .| uniq -c | sed "s/^/${BAM}/"
    }
    export -f cntrds

    ls *bam | parallel cntrds {} | tee -a read_counts2_q30.txt



# 2. 1kbp +/- TSS

### Counting reads from bam files across human reference genome for 1kbp +/- TSS to identify associated genes

```{bash}
#!/bin/bash

SAF=/group/card2/Evangelyn_Sim/Transcriptome_chromatin_human/Sequencing_ATAC_RNA/refgenome/tss_1kbp.saf
OUTPE=atac_hum_tss_pe_mapk30_q30.mx

#featureCounts -p -Q 30 -T 20 -s 0 -a $SAF -F SAF -o $OUTPE *bam


```


### Tidy 1kbp +/- TSS count matrix

```{bash}

#!/bin/bash

for MX in `ls *mx` ; do
   sed 1d $MX | sed 's/.mg.mapq30.bam//g' > $MX.all
   sed 1d $MX | cut -f1-6 | sed 's/.mg.mapq30.bam//g' > $MX.chr
   sed 1d $MX | cut -f1,7- | sed 's/.mg.mapq30.bam//g' > $MX.all.fix
   sed 1d $MX | cut -f1,7-26 | sed 's/.mg.mapq30.bam//g' > $MX.hum.fix
   
done
wait

```


### Filter out low counts genes from 1kbp +/- TSS count matrix

Filtering out low counts genes by running the following filter.sh as 

  bash filter.sh hrna_dev_mf_fulllen_se_strrev_q30.mx.all.fix

  filter.sh

    head -1 $1 > ${1}_filt
    awk '{
      min = max = sum = $2;       # Initialize to the first value (2nd field)
      sum2 = $2 * $2              # Running sum of squares
      for (n=3; n <= NF; n++) {   # Process each value on the line
        if ($n < min) min = $n    # Current minimum
        if ($n > max) max = $n    # Current maximum
        sum += $n;                # Running sum of values
        sum2 += $n * $n           # Running sum of squares
      }
      print sum/(NF-1) ;
    }' $1 > avg
    paste avg $1 | awk '$1 >= 10' | cut -f2- | tr ' ' '\t' >> ${1}_filt
    rm avg



# 3. Peaks

### Peak call of individual sample

    #!/bin/bash
    BAMS='*bam'
    BASENAME=humanATAC
    PEAKBED=${BASENAME}_peaks.bed
    PEAKSAF=${BASENAME}_peaks.saf
    OUT=${BASENAME}_pks.txt
    MX=${BASENAME}_pks_se.mx

    PATH=$PATH:/usr/local/installed/macs/1.4.2-1/python-2.7.11/.//bin/

    ls $BAMS | parallel macs14 -t {} -n {}_macs

    done

    exit


### Curate peaks that exist in more than 2 or 3 samples to form a peak set


    for BED in *peaks.bed ; do
     awk '{OFS="\t"} {if ($2<1) print $1,1,$3 ; else print $0 }' $BED | awk 'NF=="5"'> tmp
     mv tmp $BED
    done

    rm humanATAC_peaks_cov*.bed
    for COV in 2 3 ; do
      bedtools multiinter -i *_macs_peaks.bed \
     | cut -f-4 | awk -v C=$COV '$4>=C && NF==4' \
     | bedtools merge -i - > humanATAC_peaks_cov${COV}.bed
    done

    exit


### Remove blacklisted peaks


    #!/bin/bash

    wget https://github.com/Boyle-Lab/Blacklist/blob/master/lists/Blacklist_v1/hg38-blacklist.bed.gz

    cat hg38.blacklist.bed | sed 's/chr//' > hg38.blacklist.tidy.bed
    
    BL=/group/card2/Evangelyn_Sim/Transcriptome_chromatin_human/Sequencing_ATAC_RNA/refgenome/hg38.blacklist.tidy.bed

    bedops -n -1 humanATAC_peaks_cov2.bed $BL  > humanATAC_peaks_cov2_rmBL.bed

    bedops -n -1 humanATAC_peaks_cov3.bed $BL  > humanATAC_peaks_cov3_rmBL.bed


### Count individual sample to the curated peak set

```{bash}
#!/bin/bash

for BED in humanATAC*bed ; do

  SAF=$BED.saf
  OUT=$SAF.pe.q30.mx
  awk '{OFS="\t"} {print "PK"NR"_"$1":"$2"-"$3,$1,$2,$3,"+"}' $BED > $SAF
  ( featureCounts -p -Q 30 -T 20 -s 0 -a $SAF -F SAF -o $OUT *bam
  sed 1d $OUT | cut -f1,7- > tmp ; mv tmp $OUT ) &

done

```

### Tidy peak count matrix

```{bash}
#!/bin/bash

for MX in `ls *mx` ; do
   cat $MX | sed 's/.mg.mapq30.bam//g'  > $MX.fix
   cat $MX | cut -f1-21 | sed 's/.mg.mapq30.bam//g' > $MX.hum.fix
done
wait

```


### Filter out low counts genes from peak count matrix

Filtering out low counts genes by running the following filter.sh as 

  bash filter.sh hrna_dev_mf_fulllen_se_strrev_q30.mx.all.fix

  filter.sh

    head -1 $1 > ${1}_filt
    awk '{
      min = max = sum = $2;       # Initialize to the first value (2nd field)
      sum2 = $2 * $2              # Running sum of squares
      for (n=3; n <= NF; n++) {   # Process each value on the line
        if ($n < min) min = $n    # Current minimum
        if ($n > max) max = $n    # Current maximum
        sum += $n;                # Running sum of values
        sum2 += $n * $n           # Running sum of squares
      }
      print sum/(NF-1) ;
    }' $1 > avg
    paste avg $1 | awk '$1 >= 10' | cut -f2- | tr ' ' '\t' >> ${1}_filt
    rm avg


# 4. Group analysis

### Merge .bam files from the same group for transcription factor motif analysis and IGV visualization of peaks

```{bash}
#!/bin/bash

# mkdir called merged then ln all *.bam to the folder

# then run samtools merge on the bam files
samtools view -H `ls *bam | head -1` > header.sam
for BASE in `ls *bam | cut -d '_' -f1 | sort -u ` ; do
  rm $BASE.mg.bam
  samtools merge -h header.sam $BASE.mg.bam ${BASE}*bam &
done
wait



```

### Index merged .bam files

    #!/bin/bash

    # function to run bwa in paired end mode
    runbamindex() {
    BAM=$1

    samtools index $BAM
    }
    export -f runbamindex

    ls *.bam | parallel -u -j4 runbamindex {}
    
    
### MACS peak calling and then remove blacklisted peaks (method same as above)

### ### Remove blacklisted peaks

    #!/bin/bash

    BL=/group/card2/Evangelyn_Sim/Transcriptome_chromatin_human/Sequencing_ATAC_RNA/refgenome/hg38.blacklist.tidy.bed

    for BED in *_macs_peaks.bed ; do
       bedops -n -1 $BED $BL  > $BED.rmBL.bed
    done
